smoke-test/tests/actions/doc_propagation/test_propagation.py

```diff
@@ -0,0 +1,187 @@
+import logging
+import os
+import tempfile
+from pathlib import Path
+from random import randint
+from typing import Any, List
+
+import datahub.metadata.schema_classes as models
+import pytest
+from datahub.emitter.mce_builder import make_dataset_urn, make_schema_field_urn
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.ingestion.api.common import PipelineContext, RecordEnvelope
+from datahub.ingestion.api.sink import NoopWriteCallback
+from datahub.ingestion.graph.client import DatahubClientConfig, DataHubGraph
+from datahub.ingestion.sink.file import FileSink, FileSinkConfig
+from datahub.utilities.urns.urn import Urn
+
+from tests.utils import (
+    delete_urns_from_file,
+    get_gms_url,
+    ingest_file_via_rest,
+    wait_for_writes_to_sync,
+)
+
+logger = logging.getLogger(__name__)
+
+
+start_index = randint(10, 10000)
+dataset_urns = [
+    make_dataset_urn("snowflake", f"table_foo_{i}")
+    for i in range(start_index, start_index + 10)
+]
+
+
+class FileEmitter:
+    def __init__(self, filename: str) -> None:
+        self.sink: FileSink = FileSink(
+            ctx=PipelineContext(run_id="create_test_data"),
+            config=FileSinkConfig(filename=filename),
+        )
+
+    def emit(self, event):
+        self.sink.write_record_async(
+            record_envelope=RecordEnvelope(record=event, metadata={}),
+            write_callback=NoopWriteCallback(),
+        )
+
+    def close(self):
+        self.sink.close()
+
+
+def sanitize_field(field: models.SchemaFieldClass) -> models.SchemaFieldClass:
+    if field.fieldPath.startswith("[version=2.0]"):
+        field.fieldPath = field.fieldPath.split(".")[-1]
+
+    return field
+
+
+def sanitize(event: Any) -> Any:
+    if isinstance(event, MetadataChangeProposalWrapper):
+        if event.aspectName == "schemaMetadata":
+            assert isinstance(event.aspect, models.SchemaMetadataClass)
+            schema_metadata: models.SchemaMetadataClass = event.aspect
+            schema_metadata.fields = [
+                sanitize_field(field) for field in schema_metadata.fields
+            ]
+
+    return event
+
+
+def create_test_data(filename: str, test_resources_dir: str) -> List[str]:
+    def get_urns_from_mcp(mcp: MetadataChangeProposalWrapper) -> List[str]:
+        assert mcp.entityUrn
+        urns = [mcp.entityUrn]
+        if mcp.aspectName == "schemaMetadata":
+            dataset_urn = mcp.entityUrn
+            assert isinstance(mcp.aspect, models.SchemaMetadataClass)
+            schema_metadata: models.SchemaMetadataClass = mcp.aspect
+            for field in schema_metadata.fields:
+                field_urn = make_schema_field_urn(dataset_urn, field.fieldPath)
+                urns.append(field_urn)
+        return urns
+
+    from datahub.api.entities.dataset.dataset import Dataset
```

Comment:
why are these imports stuffed inside the methods? unless there's a good reason, they should be at the top of the file